packages/@expo/config-plugins/src/android/Permissions.ts

```diff
@@ -24,15 +24,15 @@ export const withPermissions: ConfigPlugin<string[] | void> = (config, permissio
 
 /** Given a permission or list of permissions, block permissions in the final `AndroidManifest.xml` to ensure no installed library or plugin can add them. */
 export const withBlockedPermissions: ConfigPlugin<string[] | string> = (config, permissions) => {
-  const resolvedPermissions = (Array.isArray(permissions) ? permissions : [permissions]).filter(
-    Boolean
+  const resolvedPermissions = prefixAndroidPermissionsIfNecessary(
+    (Array.isArray(permissions) ? permissions : [permissions]).filter(Boolean)
   );
 
   if (config?.android?.permissions && Array.isArray(config.android.permissions)) {
     // Remove any static config permissions
-    config.android.permissions = config.android.permissions.filter(
-      (permission) => !resolvedPermissions.includes(permission)
-    );
+    config.android.permissions = prefixAndroidPermissionsIfNecessary(
```

Comment:
This doesn't seem exactly right. But maybe it's fine, I'm not sure if the original pass is really needed.

Reply:
I did this to "filter" mixed permission notations. Basically, when someone uses:

```json
{
  "android": {
    "permissions": ["CAMERA"]
  }
}
```

and has `withBlockedPermissions(..., ["android.permissions.CAMERA"])` we need to sanitize the mixed set. (Also applies to the other way around)